Spigot-Server-Patches/0332-Optimize-World-search-radius-functions.patch

```diff
@@ -0,0 +1,215 @@
+From 3ec6476aeadd6d74b334f0d5e7b1ee14bdc0d3c7 Mon Sep 17 00:00:00 2001
+From: egg82 <phantom_zero@ymail.com>
+Date: Tue, 7 Aug 2018 17:55:35 -0600
+Subject: [PATCH] Optimize World search/radius functions
+
+Specifically fixes #1289
+
+Affected functions:
+World.a(double d0, double d1, double d2, double d3, Predicate<Entity> predicate)
+World.isPlayerNearby(double d1, double d2, double d3, double searchRadius)
+World.a(double d0, double d1, double d2, double d3, double d4, @Nullable Function<EntityHuman, Double> function, @Nullable Predicate<EntityHuman> predicate)
+
+Renamed variables in functions and headers, and completely replaced said functions
+The rename shouldn't affect anything but readability
+
+The new code utilizes CraftWorld.getNearbyEntities for later optimization in a future patch
+It shouldn't affect previous behavior and in fact makes attempts at heading off edge-cases early
+
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index 004c3ec4..1307e503 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -24,6 +24,7 @@ import com.google.common.collect.Maps;
+ import java.util.HashMap; // Paper
+ import java.util.Map;
+ import org.bukkit.Bukkit;
++import org.bukkit.Location; // Paper
+ import org.bukkit.block.BlockState;
+ import org.bukkit.craftbukkit.CraftServer;
+ import org.bukkit.craftbukkit.CraftWorld;
+@@ -2575,7 +2576,8 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
+         return i;
+     }
+ 
+-    @Nullable
++    // Paper start
++    /*@Nullable
```

Comment:
Cant do this, just delete the lines of code instead.
This will cause headaches in maintenance.

I'm ok with the var name change, just don't comment out old code, remove it instead, as the new code MUST replace the old method so it conflicts (conflicts are good)